### ISSUE TITLE
Some issue with AbstractWriter and temporary folder

### ISSUE THREAD

**Issue by martinsuly** [PHPOffice/PHPWord#858]
I am not sure if there is some configuration trick for this but I found some issue with AbstractWriter  (PhpWord/Writer/AbstractWriter.php) and the name of temporary location where it saves files. It happens when more concurrent processes use PHPWord - in same time trying to create for example a word file. In the default configuration it uses a temporary location "/tmp/PHPWordWriter" and at the end of generation it does a clean up action for this folder in method "deleteDir" of this class. 

Let's imagine we had two processes which needed to generate some word files and started in same time. The first one didn't have so much work, so it ended sooner than the second one and it called method "deleteDir". All files in folder "/tmp/PHPWordWriter" have been deleted. But the second process had some files here too so this second process couldn't reach its temporary files and couldn't finish generation of the document.

This is some workaround that works for me:

```
$dirname = uniqid('', true);
$dir = \PhpOffice\PhpWord\Settings::getTempDir() . '/' . $dirname;

mkdir($dir);
\PhpOffice\PhpWord\Settings::setTempDir($dir);

$obj_writer = \PhpOffice\PhpWord\IOFactory::createWriter($phpword, 'Word2007');
$obj_writer->save('test.docx');

rmdir($dir);
```

**Comment by troosan** [comment]
fixed in 0.14

**Comment by nicoder** [comment]
in commit 512cf952aeaaa9badcce3c0eb88114f6beaeeb22?
